src/alchemlyb/postprocessors/units.py

```diff
@@ -0,0 +1,91 @@
+'''Physical and mathematical constants and units.'''
+# Taken from scipy.constants since py2 doesn't support it
+#: Boltzmann's constant :math:`k_B` in kJ/(mol K); value from `NIST CODATA: k<https://physics.nist.gov/cgi-bin/cuu/Value?k>`_.
+Boltzmann_constant = 1.380649e-23
+#: Avogadro constant :math:`N_A` in 1/mol; value from `NIST CODATA: k<https://physics.nist.gov/cgi-bin/cuu/Value?na|search_for=avogadro>`_.
+Avogadro_constant = 6.02214076e+23
+kJ2kcal = 0.239006
+
+
+def to_kT(df):
+    """Convert the unit of a DataFrame to `kT`. The DataFrame need to have
+    attribute `temperature` and `energy_unit`.
+
+    Parameters
+    ----------
+    df : DataFrame
+        DataFrame to convert unit.
+
+    Returns
+    -------
+    DataFrame
+        `df` converted.
+    """
+    assert 'temperature' in df.attrs, 'Attribute temperature not found in the' \
+                                      ' input Dataframe.'
+    assert 'energy_unit' in df.attrs, 'Attribute energy_unit not found in the' \
+                                      ' input Dataframe.'
```

Comment:
Replace `assert` with raising a `TypeError`, saying that we need a df with attrs according to https://alchemlyb.readthedocs.io/en/stable/parsing.html#standard-forms-of-raw-data . Ultimately, the dfs are user input so we have to check them properly.

(`assert` can be optimized away and should not be used to check user input.)